tremor-codec/src/codec/graphite.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2024, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The `graphite-plaintext` codec supports the [graphite plaintext format](https://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-plaintext-protocol).
+//!
+//! The format is similar to the `influx` line protocol.
+//!
+//! The codec translates a single `graphite` measurement line into a structured event.
+//!
+//! ## Example
+//!
+//! The native format of a single graphite event line is as follows:
+//!
+//! ```text
+//! <metric_path> <value> <timestamp>
+//! ```
+//!
+//! The equivalent representation as a tremor value:
+//!
+//! ```json
+//! {
+//!   "metric": "my.metric.path",
+//!   "value": 7,
+//!  "timestamp": 1620649445 // number of seconds since the epoch
```

Comment:
looking at the code multiplying makes sense, if -1 is passed it already uses `ns` instead of `s`

```rs
    if "-1" == ts {
        m.insert("timestamp".into(), Value::from(ingest_ns));
    } else {
        let ts = lexical::parse::<u64, _>(ts)?;
        m.insert("timestamp".into(), Value::from(ts));
    }
```

Reply:
I'll do this separately as it deserves changes to the tests

Reply:
I also added comments as this isn't obvious to anyone casually looking at this. Especially our future selves...

Reply:
Good call, it'll help us down the road :D